MarkEditMac/Sources/Editor/Models/EditorDocument.swift

```diff
@@ -225,12 +225,44 @@ extension EditorDocument {
   }
 
   override func writableTypes(for saveOperation: NSDocument.SaveOperationType) -> [String] {
+    // Include all markdown and plaintext types, but prioritize the configured default
+    var exportTypes = NewFilenameExtension.allCases
+      .sorted { lhs, _ in
+        lhs.rawValue == AppPreferences.General.newFilenameExtension.rawValue
+      }
+      .map { $0.exportedType }
+
     // Enable *.textbundle only when we have the bundle, typically for a duplicated draft
-    textBundle == nil ? [AppPreferences.General.newFilenameExtension.exportedType] : ["org.textbundle.package"]
+    if textBundle != nil {
+      exportTypes.insert("org.textbundle.package", at: 0)
+    }
+
+    return exportTypes
   }
 
   override func fileNameExtension(forType typeName: String, saveOperation: NSDocument.SaveOperationType) -> String? {
-    typeName.isTextBundle ? "textbundle" : AppPreferences.General.newFilenameExtension.rawValue
+    typeName.isTextBundle ? "textbundle" : NewFilenameExtension.allCases.first { $0.exportedType == typeName }?.rawValue
+  }
+
+  /// Catch invalid output paths early so we can present more informative errors
+  @objc override func handleSave(_ command: NSScriptCommand) -> Any? {
```

Comment:
Is `@objc` required?

Reply:
Also, I am assuming this will only be called from AppleScript, doesn't affect normal saving process at all, right?

Asking this because the normal saving process is already a mess.

Reply:
I think we can add a

```swift
var isTextBundle: Bool {
  textBundle != nil
}
`

We can use this to replace all `textBundle != nil` and move the `handleSave(_ command: NSScriptCommand)` to `EditorDocument+Scripting`, what do you think?